modules/tensor_mechanics/include/materials/UpdateReturnMappingStressBase.h

```diff
@@ -0,0 +1,46 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+#ifndef UPDATERETURNMAPPINGSTRESSBASE_H
+#define UPDATERETURNMAPPINGSTRESSBASE_H
+
+#include "RankTwoTensor.h"
+#include "RankFourTensor.h"
+#include "Conversion.h"
+
+//Forward declaration
+class UpdateReturnMappingStressBase;
+
+template<>
+InputParameters validParams<UpdateReturnMappingStressBase>();
+
+/**
+ * UpdateReturnMappingStressBase is a material that is not called by MOOSE because
+ * of the compute=false flag set in the parameter list.  This class is a base class
```

Comment:
I'd reword the sentence: 
"This class is a base class for materials which perform internal Newton interations, such as radial return mapping materials."
to something like:
"This class is a base class for materials that use Newton iterations to perform radial return mapping iterations to compute inelastic strains."